src/app/activity/activity.service.ts

```diff
@@ -111,7 +114,11 @@ export class ActivityService {
             type: 'Assessment',
             contextId: contextIds[element[element.model].id] || 0,
             loadingStatus: true,
-            isForTeam: element[element.model].is_team
+            isForTeam: element[element.model].is_team,
+            // @TODO need to update after api ready
+            dueDate: this.utils.validateDueDates(element[element.model].deadline),
+            isOverdue: this.utils.validateDueDates(element[element.model].deadline).indexOf('Overdue') !== -1 ? true : false,
```

Comment:
`isOverdue` can be checked from `this.utils.timeComparer()`, better to not check the specific string 'Overdue'